### DISCUSSION TITLE
I'm trying to understand the meaning of batch_size, n_batches, and probability

### DISCUSSION THREAD

**Discussion by romankempt** [deepmodeling/deepmd-kit#1989] [Q&A]
I don't quite grasp what the importance of those parameters is, but observe that either setting them to "auto" or "1" (as in the Mg-Y-alloy tutorial) heavily influences the training results.

Let's say I have three data systems with 3, 6, and 9 atoms, containing 33, 33, and 34 frames (adding up to 100).

If I set the batch_size to 1, should the number of batches n_bch not correspond to the number of frames? e.g.:
```
system  natoms  bch_sz   n_bch   prob  pbc 
a          3             1           33        0.33   T
b          6             1           33        0.33   T
c          9              1          34        0.34   T
```

If I set it to "auto:32", then what does the batch_sz determine? What does n_bch do?
e.g.:
```
system  natoms  bch_sz   n_bch   prob  pbc 
a          3             11          ?        ?   T
b          6             6            ?        ?   T
c          9              4           ?        ?   T
```

**Comment by njzjz** [comment]
See https://docs.deepmodeling.com/projects/deepmd/en/v2.1.5/train/train-input.html#argument:training/validation_data/batch_size

In your two cases, the first batch size will be always 1, and the second equals to $\lceil \frac{32}{N} \rceil$.

**Reply by romankempt** [reply]
Could you elaborate further? Does the batch_sz then correspond to the number of sets?

**Reply by njzjz** [reply]
$N$ is the number of atoms.
